changelog_gen/extractor.py

```diff
@@ -21,37 +20,66 @@ def __init__(self: typing.Self, supported_sections: list[str], *, dry_run: bool
         self.dry_run = dry_run
         self.supported_sections: dict[str, str] = supported_sections
 
-        if not self.release_notes.exists() or not self.release_notes.is_dir():
-            msg = "No release notes directory found."
-            raise errors.NoReleaseNotesError(msg)
+        self.has_release_notes = self.release_notes.exists() and self.release_notes.is_dir()
 
     def extract(self: typing.Self, section_mapping: dict[str, str] | None = None) -> SectionDict:
         """Iterate over release note files extracting sections and issues."""
         section_mapping = section_mapping or {}
 
-        sections = defaultdict(OrderedDict)
-
-        # Extract changelog details from release note files.
-        for issue in sorted(self.release_notes.iterdir()):
-            if issue.is_file and not issue.name.startswith("."):
-                issue_ref, section = issue.name.split(".")
-                section = section_mapping.get(section, section)
-
-                breaking = False
-                if section.endswith("!"):
-                    section = section[:-1]
-                    breaking = True
-
-                contents = issue.read_text().strip()
-                if section not in self.supported_sections:
-                    msg = f"Unsupported CHANGELOG section {section}"
-                    raise errors.InvalidSectionError(msg)
+        sections = defaultdict(dict)
+
+        if self.has_release_notes:
+            # Extract changelog details from release note files.
+            for issue in sorted(self.release_notes.iterdir()):
+                if issue.is_file and not issue.name.startswith("."):
+                    issue_ref, section = issue.name.split(".")
+                    section = section_mapping.get(section, section)
+
+                    breaking = False
+                    if section.endswith("!"):
+                        section = section[:-1]
+                        breaking = True
+
+                    contents = issue.read_text().strip()
+                    if section not in self.supported_sections:
+                        msg = f"Unsupported CHANGELOG section {section}"
+                        raise errors.InvalidSectionError(msg)
+
+                    sections[section][issue_ref] = {
+                        "description": contents,
+                        "breaking": breaking,
+                    }
```

Comment:
🙃 : make a method for that?

Reply:
Yeah, this got refactored later on I believe (will link PR)

Reply:
https://github.com/EdgyEdgemond/changelog-gen/pull/77/files#diff-5ec09b87a087c81907fc97531029c0a529055bc916aa368dacc36cfb429fe67aR37